src/components/form-helpers/form-helpers.js

```diff
@@ -90,20 +90,31 @@ const _FormWithCaptcha = ({
   dependencies: { captchaUtilsService },
   ...rest
 }) => {
+  let timer1;
+  let timer2;
+  let timer3;
   /** Store original onSubmit because I need to replace it with verifyCaptchaAndSubmit */
   const originalOnSubmit = onSubmit;
   const [Captcha, verifyCaptcha] = captchaUtilsService.useCaptcha();
 
+  useEffect(() => {
+    return () => {
+      clearTimeout(timer1);
+      clearTimeout(timer2);
+      clearTimeout(timer3);
+    };
+  }, []);
```

Comment:
There should be a cleaner way to do this.

Reply:
Maybe a kind of hook? something like `useTimeout()` that avoids effects when the component has been unloaded?

Reply:
Maybe, something like this? 

**const createTimeout = useTimeout();**

inside the function it would be used as:

const verifyCaptchaAndSubmit = async (values, formikProps) => {
      createTimeout(() => formikProps.setSubmitting(true), captchaDelay);

      createTimeout((() => formikProps.setSubmitting(true), captchaDelay);

      createTimeout(() => formikProps.setSubmitting(false), captchaDelay);
}

Reply:
What about think ?

Reply:
Looks nice! 

Does `useTimeout` already exist or you will create it?

Reply:
I have to create it

Reply:
I think that we can have this done in a separate PR.